ui/app/templates/components/oidc/provider-form.hbs

```diff
@@ -0,0 +1,89 @@
+<PageHeader as |p|>
+  <p.top>
+    <nav class="breadcrumb">
+      <ul>
+        <li>
+          <span class="sep">&#x0002f;</span>
+          {{#if @model.isNew}}
+            <LinkTo @route="vault.cluster.access.oidc.providers">
+              Providers
+            </LinkTo>
+          {{else}}
+            <LinkTo @route="vault.cluster.access.oidc.providers.provider.details" @model={{@model.name}}>
+              Details
+            </LinkTo>
+          {{/if}}
+        </li>
+      </ul>
+    </nav>
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3" data-test-oidc-provider-title>
+      {{if @model.isNew "Create" "Edit"}}
+      provider
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+<form {{on "submit" (perform this.save)}}>
+  <div class="box is-sideless is-fullwidth is-bottomless">
+    <MessageError @errorMessage={{this.error}} class="has-top-margin-s" />
+    {{#each @model.formFields as |attr|}}
+      <FormField data-test-field={{true}} @attr={{attr}} @model={{@model}} @modelValidations={{this.modelValidations}} />
+    {{/each}}
+  </div>
+  {{! RADIO CARD + SEARCH SELECT }}
+  <div class="box is-sideless is-fullwidth is-marginless has-top-padding-xxl">
+    <h4 class="title is-4">Allowed applications</h4>
+    <div class="is-flex-row">
+      <RadioCard
+        @title="Allow every application to use"
+        @description="All applications can use this provider for authentication requests."
+        @icon="globe"
+        @value="allow_all"
+        @groupValue={{this.radioCardGroupValue}}
+        @onChange={{this.handleClientSelection}}
+      />
+      <RadioCard
+        @title="Limit access to selected application"
+        @description="Only selected applications can use this provider for authentication requests."
+        @icon="globe-private"
+        @value="limited"
+        @groupValue={{this.radioCardGroupValue}}
+        @onChange={{this.handleClientSelection}}
+      />
+    </div>
+    {{#if (eq this.radioCardGroupValue "limited")}}
+      <SearchSelect
+        @id="allowedClientIds"
+        @subLabel="Application name"
+        @models={{array "oidc/client"}}
+        @inputValue={{@model.allowedClientIds}}
+        @onChange={{this.handleClientSelection}}
+        @disallowNewItems={{true}}
+        @fallbackComponent="string-list"
```

Comment:
I tested this out. I created a new client and typed in the assignment name and when I said create, it failed to redirect to the details view. My initial thought is it's because we aren't setup to manually enter an assignment name? Or it's because we're not handling the group and entity search properly? Here's a mov of what I did (checked out on this branch).

Reply:
https://user-images.githubusercontent.com/6618863/183536833-9e04cd5e-c1a7-485b-8714-e29f3c79daee.mov

Reply:
Yeah the modal pops up just to create an assignment - on create the modal closes, and the flash message "Successfully created an assignment" displays. Then you have to click "Create" on the client form to complete creating the application itself. 😄

Reply:
Also this .mov is from the create/edit application form - not the provider form this PR covers. Just fyi!

Reply:
Wow, I missed that (the modal closed and I thought I was done as a user) and I missed the wrong form 🤦🏻‍♀️. I wonder if anyone else falls into the same patter I did or if it is just how I move through forms. Maybe it's because they both say "Create" to complete the step. The UX is to click create twice. Did you ever slip up there, or was the flow natural?

Reply:
@yankun91 - thoughts on this? I didn't have any mixup between the create form for the assignment modal and the one for the client (but I developed it! So I knew the flow 😅 )

Reply:
Yeah, I can see where the confusion comes from. I roughly remembered that we had a conversation around whether the button in the modal should be 'save' or 'create'. Would 'save' work better here?

Reply:
Yeah in the RFC it was suggested changing it from "Save" to "Create". Something to keep in mind - the modal form is the _same_ as the one in the Assignment Create route - so any changes here will be for both form. 

I think we should keep as is right now, but I'm putting together a bug bash template and will make this a task for folks to do so we can get some more feedback.